fedimint-client/src/lib.rs

```diff
@@ -564,13 +564,26 @@ const SUPPORTED_CORE_API_VERSIONS: &[fedimint_core::module::ApiVersion] =
 pub type ModuleGlobalContextGen = ContextGen<DynGlobalClientContext>;
 
 /// Resources particular to a module instance
-pub struct ClientModuleInstance {
+pub struct ClientModuleInstance<'m, M: ClientModule> {
     /// Instance id of the module
     pub id: ModuleInstanceId,
     /// Module-specific DB
     pub db: Database,
     /// Module-specific API
     pub api: DynModuleApi,
+
+    module: &'m M,
+}
+
+impl<'m, M> ops::Deref for ClientModuleInstance<'m, M>
+where
+    M: ClientModule,
+{
+    type Target = M;
+
+    fn deref(&self) -> &Self::Target {
+        self.module
+    }
 }
```

Comment:
Feels a bit dirty but I like it xD

Reply:
We could just make it a method or a field, and it wouldn't be that much of a downgrade.

... but if we squint our eyes, we can consider it a smart pointer over module, so I guess it's OK?